pkg/controllers/execution/execution_controller.go

```diff
@@ -100,9 +101,8 @@ func (c *Controller) Reconcile(ctx context.Context, req controllerruntime.Reques
 // SetupWithManager creates a controller and register to controller manager.
 func (c *Controller) SetupWithManager(mgr controllerruntime.Manager) error {
 	return controllerruntime.NewControllerManagedBy(mgr).
-		For(&workv1alpha1.Work{}).
+		For(&workv1alpha1.Work{}, builder.WithPredicates(c.PredicateFunc)).
 		WithEventFilter(predicate.GenerationChangedPredicate{}).
```

Comment:
Should we do the same thing for  `predicate.GenerationChangedPredicate{}`?

Reply:
`GenerationChangedPredicate` could be a global predicate function. It can take effect when any `watched` resource changed. Well, it does not matter here cause we don't have any other resources to watch.

Reply:
Yes, no big deal.